docs/en/interfaces/predefined_http.md

```diff
@@ -0,0 +1,216 @@
+
+Example:
+
+```xml
+<http_handlers>
+    <replicas_status_handler>/replicas_status</replicas_status_handler>
+</http_handlers>
+```
+
+```bash
+$ curl 'http://localhost:8123/replicas_status'
+Ok.
+```
+
+## predefined_query_handler
+
+You can configure `<method>`, `<headers>`, `<url>` and `<queries>` in `<predefined_query_handler>`.
+
+`<method>` is responsible for matching the method part of the HTTP request. `<method>` fully conforms to the definition of method in the HTTP protocol. It is an optional configuration.
+
+`<url>`  is responsible for matching the url part of the HTTP request. It is compatible with RE2's regular expressions. It is an optional configuration.
+
+`<headers>` is responsible for matching the header part of the HTTP request. It is compatible with RE2's regular expressions. It is an optional configuration.
+
+`<queries>` value is a predefined query of `<predefined_query_handler>`, which is executed by ClickHouse when an HTTP request is matched and the result of the query is returned. It is a must configuration.
+
+`<predefined_query_handler>` supports setting Settings and query_params values. 
+
+To experiment with this functionality, the example defines the values of max_threads and max_alter_threads and queries whether the Settings were set successfully.
+
+Example:
+
+```xml
+<root_handlers>
+	<predefined_query_handler>
+	    <method>GET</method>
+	    <headers>
+	        <XXX>TEST_HEADER_VALUE</XXX>
+	        <PARAMS_XXX><![CDATA[(?P<name_1>[^/]+)(/(?P<name_2>[^/]+))?]]></PARAMS_XXX>
+	    </headers>
+	    <url><![CDATA[/query_param_with_url/\w+/(?P<name_1>[^/]+)(/(?P<name_2>[^/]+))?]]></url>
+	    <queries>
+	        <query>SELECT value FROM system.settings WHERE name = {name_1:String}</query>
+	        <query>SELECT name, value FROM system.settings WHERE name = {name_2:String}</query>
+	    </queries>
+	</predefined_query_handler>
+</root_handlers>
+```
+
+```bash
+$ curl -H 'XXX:TEST_HEADER_VALUE' -H 'PARAMS_XXX:max_threads' 'http://localhost:8123/query_param_with_url/1/max_threads/max_alter_threads?max_threads=1&max_alter_threads=2'
+1
+max_alter_threads	2
+```
+
+> Note:
+>
+> In one `<predefined_query_handler>`, one `<queries>` only supports one `<query>` of an insert type.
+
+## dynamic_query_handler
+
+`<dynamic_query_handler>` than `<predefined_query_handler>` increased  `<query_param_name>` .
+
+ClickHouse extracts and executes the value corresponding to the `<query_param_name>` value in the url of the HTTP request.
+ClickHouse default setting `<query_param_name>` is `/query` . It is an optional configuration.
+
+To experiment with this functionality, the example defines the values of max_threads and max_alter_threads and queries whether the Settings were set successfully.
```

Comment:
* same issue as above
* also it doesn't explain the specifics of this example comparing to previous one